src/Controller/ResourceController.php

```diff
@@ -345,27 +364,41 @@ private function getDirectory(Request $request, $path, $responseFormat, array $v
             ];
         }
 
-        $content = $this->getGraphForPath($request, $path)->serialise($responseFormat, $options);
+        $content = $this->getGraphForPath($request, $requested_path)->serialise($responseFormat, $options);
 
         $headers = [
-            "Last-Modified" => $modifiedTime->format(\DateTime::W3C),
-            "Link" => ["<{$this->LDP_NS}Resource>; rel=\"type\"",
-                       "<{$this->LDP_NS}BasicContainer>; rel=\"type\""],
-            "Vary" => "Accept",
+            "Link" => ["<".self::LDP_NS."Resource>; rel=\"type\"",
+                       "<".self::LDP_NS."BasicContainer>; rel=\"type\""],
             "Content-Type" => $responseMimeType,
             "Content-Length" => strlen($content)
         ];
 
-        if (!$doGet) {
-            $content = '';
+        if ($request->getMethod() == 'GET') {
+            $response->setContent($content);
         }
-        return new Response($content, 200, $headers);
+        $response->headers->add($headers);
+        return $response;
+    }
+
+    /**
+     * Simple Convulsive Etag generator for files
+     *
+     * @return array
+     */
+    private function provideEtag()
```

Comment:
@acoburn the reason i generated that weird provideEtag one is because i never know/i'm pretty sure there will be cases were i would end needing to generated the Etag (like in the basic container) or i will just blindly would want to reuse the one created but doing as i want a new one. Adding private properties to a controller is not standard really, but since the controller is born and dies with the request, and nothing fancy is moved around i used that as a persistant state helper for request basic headers.